app/scripts/controllers/service.js

```diff
@@ -60,6 +99,36 @@ angular.module('openshiftConsole')
           }
         );
 
+        watches.push(DataService.watch("services", context, function(services) {
+          $scope.services = services.by("metadata.name");
+        }));
+
+        watches.push(DataService.watch("pods", context, function(pods) {
+          $scope.pods = [];
+          angular.forEach(pods.by("metadata.name"), function(pod) {
+            $scope.pods.push(pod);
+          });
+          getPodsForService($scope.service, $scope.pods);
+        }));
+
+        watches.push(DataService.watch("endpoints", context, function(endpoints) {
+          $scope.podsWithEndpoints = {};
+          var svcEndpoint  = endpoints.by("metadata.name");
```

Comment:
i would probably just make this a one-liner instead of initializing a variable as a map and then turning it into a singleton
```
var svcEndpoint = endpoints.by("metadata.name")[$routeParams.service];
```